core/state/shared_pool.go

```diff
@@ -0,0 +1,36 @@
+package state
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"sync"
+)
+
+// sharedStorage is used to store maps of originStorage of stateObjects
+type SharedStorage struct {
+	*sync.RWMutex
+	sharedMap map[common.Address]*sync.Map
+}
+
+func NewSharedStorage() *SharedStorage {
+	sharedMap := make(map[common.Address]*sync.Map, 1500)
```

Comment:
Why `1500` is used here?  `1500` is not small size,  it means for testnet which has less transactions, it will always allocate much more memory for each block.

Reply:
yes, I has not considered the waste of memory allocation.   I set the value beacause  most of the blocks does not have more than 1200 stateObjects in mainnet

Reply:
updated